core/commands/repo.go

```diff
@@ -302,13 +302,35 @@ var repoVerifyCmd = &oldcmds.Command{
 
 		var fails int
 		var i int
+		var count int
+		verify := make(chan string)
+		defer close(verify)
+
 		for k := range keys {
-			_, err := bs.Get(k)
-			if err != nil {
+			go func() {
```

Comment:
So, spawning a goroutine per key is going to blow through memory like nothing else. We'll have to:

1. Span N goroutines where N is something like `runtime.NumCPU * 2` (ish?).
2. Have them all read off the keys channel.
3. Have them all write to the same "results" channel.
4. Have them call `Done` on a wait-group on exit.
5. Spin up another goroutine that closes the "results" channel when the wait-group finishes.
6. Outside, have a for loop reading off the results channel.

Reply:
This is great advice. Thx a lot. :)